environment/src/test/java/com/sequenceiq/environment/environment/encryption/EnvironmentEncryptionServiceTest.java

```diff
@@ -180,6 +180,84 @@ void testCreateEncryptionResourcesCreationRequestShouldReturnWithANewEncryptionR
+        assertEquals(creationRequest.isSingleResourceGroup(), Boolean.FALSE);
+        verifyCloudContext(creationRequest.getCloudContext());
+        assertThat(creationRequest.getCloudCredential()).isSameAs(cloudCredential);
+        assertThat(creationRequest.getId()).isEqualTo("randomGeneratedResource");
+        assertThat(creationRequest.getTags()).isSameAs(tags);
+    }
+
+    @Test
+    void testCreateEncryptionResourcesCreationRequestShouldSetIsSingleResourceGroupToTrueWhenSingleResourceGroup() {
+        EnvironmentDto environmentDto = EnvironmentDto.builder()
+                .withResourceCrn(ENVIRONMENT_CRN)
+                .withId(ENVIRONMENT_ID)
+                .withName(ENVIRONMENT_NAME)
+                .withCloudPlatform(CLOUD_PLATFORM)
+                .withCredential(credential)
+                .withLocationDto(LocationDto.builder().withName(REGION).build())
+                .withParameters(ParametersDto.builder()
+                        .withAzureParameters(AzureParametersDto.builder()
+                                .withEncryptionParameters(AzureResourceEncryptionParametersDto.builder()
+                                        .withEncryptionKeyUrl(KEY_URL)
+                                        .withEncryptionKeyResourceGroupName(KEY_URL_RESOURCE_GROUP_NAME)
+                                        .build())
+                                .withResourceGroup(AzureResourceGroupDto.builder()
+                                        .withResourceGroupUsagePattern(ResourceGroupUsagePattern.USE_SINGLE)
+                                        .withName(RESOURCE_GROUP_NAME).build())
+                                .build())
+                        .build())
+                .withCreator(USER_NAME)
+                .withAccountId(ACCOUNT_ID)
+                .build();
+        when(credentialToCloudCredentialConverter.convert(credential)).thenReturn(cloudCredential);
+        Map<String, String> tags = Map.ofEntries(entry("tag1", "value1"), entry("tag2", "value2"));
+        when(environmentTagProvider.getTags(environmentDto, ENVIRONMENT_CRN)).thenReturn(tags);
+
+        DiskEncryptionSetCreationRequest creationRequest = underTest.createEncryptionResourcesCreationRequest(environmentDto);
+
+        assertEquals(creationRequest.getEncryptionKeyUrl(), KEY_URL);
+        assertEquals(creationRequest.getDiskEncryptionSetResourceGroupName(), RESOURCE_GROUP_NAME);
+        assertEquals(creationRequest.getEncryptionKeyResourceGroupName(), KEY_URL_RESOURCE_GROUP_NAME);
+        assertEquals(creationRequest.isSingleResourceGroup(), Boolean.TRUE);
+        verifyCloudContext(creationRequest.getCloudContext());
+        assertThat(creationRequest.getCloudCredential()).isSameAs(cloudCredential);
+        assertThat(creationRequest.getId()).isEqualTo("randomGeneratedResource");
+        assertThat(creationRequest.getTags()).isSameAs(tags);
+    }
+
```

Comment:
Please add a new `createEncryptionResourcesDeletionRequest()` test case for the Multi-RG scenario, including the check for the two `CloudResource` instances being returned in `DiskEncryptionSetDeletionRequest`.